### DISCUSSION TITLE
Do you plan to add Python to Unit Test Generator?

### DISCUSSION THREAD

**Discussion by s2terminal** [keploy/keploy#2466] [General]
I am interested in Keploy's Unit Test Generator (ut-gen). [The documentation](https://keploy.io/docs/running-keploy/unit-test-generator/) says that ut-gen supports Go and Node.js.

When I try to use Python with ut-gen, I get an error saying that it is an unsupported language.
However, from [the implementation of pkg/service/utgen](https://github.com/keploy/keploy/blob/c5dac5f49124e9dbb42562bf7054d433625bd2a9/pkg/service/utgen/injector.go#L50), it looks like the implementation for Python is included.

Do you have any plans to add support for languages such as Python with ut-gen? What are the difficulties involved in adding support for Python with ut-gen?

**Comment by slayerjain** [comment]
Hi @s2terminal, I think its because we might not have tested the open source version with Python yet. Would be good to have it. 

We are also in the process of launching a self managed version of the unit test generator behind a VS  code extension that will also support Python. 

The vision is to ensure technical validation (build, coverage addition,  flakiness) of tests are done over whatever the LLM creates and also perform tweaks to src code (if necessary) to make it more testable.

Please feel free to hack around the src code and raise a PR for Python 😄

Ref: vs code extension - https://marketplace.visualstudio.com/items?itemName=Keploy.keployio

**Reply by s2terminal** [reply]
Thank you. I have created the following pull request. https://github.com/keploy/keploy/pull/2475

**Comment by s2terminal** [comment]
https://github.com/keploy/keploy/pull/2475

This pull request has been merged. This issue has been resolved. Thanks for merging this, especially during the holidays!

**Reply by nehagup** [reply]
Thank you! for adding the support and making it better. :) 
Happy holidays! 🧡
